xarray/core/dataarray.py

```diff
@@ -3999,6 +3999,56 @@ def to_dict(self, data: bool = True, encoding: bool = False) -> dict[str, Any]:
             d["encoding"] = dict(self.encoding)
         return d
 
+    def data_to_list_save_memory(self):
+        """
+        Covert this Xarray.DataArray to a list while freeing up memory as it rids of the original DataArray
+        Useful when returning data values in json format of large datasets after dataset no longer in use to avoid memory issues
+        More effecient than to_dict() regarding memory usage
+
+        Returns
+        --------
+        lst_to_return: list
+        """
+        lst_to_return = []
+        if 0 in self.shape:
+            return lst_to_return
+
+        the_data = self.data.values
+
+        for i in range(len(the_data)):
+            lst_to_return.append(the_data[0].tolist())
```

Comment:
It's worth noting that this is basically the implementation of `numpy.ndarray.tolist`, except `tolist`uses recursion in C and does not perform any memory copies (see `PyArray_ToList`).